base/number.jl

```diff
@@ -3,6 +3,15 @@
 ## generic operations on numbers ##
 
 isinteger(x::Integer) = true
+iszero(x::Number) = x == 0 # faster than x == zero(x) for complex, bigfloat, etc.
```

Comment:
To handle unitful `<:Number`s correctly, perhaps
```julia
iszero(x::Number) = x == zero(x)
```
with specializations
```julia
iszero{T<:Complex}(x::T) = x == 0
iszero(x::BitFloat) = x == 0
iszero(x::BigInt) = x == 0
```
in appropriate files for performance?